src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,259 @@
+      backdrop.on('click', () => this.close());
+      this.element.parentElement.appendChild(backdrop.el);
+    } else {
+      const dropdown = $$(document.body).find('.coveo-dropdown-background');
+      $$(dropdown).on('click', () => this.close());
+    }
+  }
+
+  private close() {
+    $$(this.element).removeClass('coveo-checkbox-container-expanded');
+    this.checkboxContainer.removeClass('coveo-checkbox-container-expanded');
+    if ($$(this.element.parentElement).find('.coveo-dropdown-background') != null && $$(document.body).find('.coveo-checkbox-container-expanded') == null) {
+      $$(this.element.parentElement).find('.coveo-dropdown-background').remove();
+    }
+  }
+
+  private handleCheckboxToggle() {
+    this.circleElement.text(this.getSelectedValues().length.toString());
+    const title = $$(this.element).find('.coveo-simplefilter-selecttext');
+    if (this.getSelectedValues().length == 1) {
+      $$(title).text(l(this.getSelectedValues()[0]));
+    } else {
+      $$(title).text(this.options.title);
+    }
+    this.queryController.executeQuery();
+  }
+
+  private createCheckboxes(value: string) {
+    const checkbox = new Checkbox(() => {
+      this.handleCheckboxToggle();
+    }, l(this.getValueCaption(value)));
+    return { checkbox, value };
+  }
+
+  public getValueCaption(value: string): string {
+    let ret = value;
+
+    if (Utils.exists(this.options.valueCaption)) {
+      if (typeof this.options.valueCaption == 'object') {
+        ret = this.options.valueCaption[ret] || ret;
+      }
+    }
+    return ret;
+  }
+
+  public getSelected(): { checkbox: Checkbox, value: string }[] {
+    return _.filter(this.checkboxes, (object: { checkbox: Checkbox, value: string }) => object.checkbox.isSelected());
+  }
+
+  public getSelectedValues(): string[] {
+    return _.map(this.getSelected(), (object: { checkbox: Checkbox, value: string }) => object.value);
+  }
+
+  private buildCircleElement(): Dom {
+    this.circleElement = $$('span', { className: 'coveo-circle' }, this.getSelectedValues().length.toString());
+    return this.circleElement;
+  }
+
+  private createWrapper() {
+    if ($$(document.body).find('.coveo-filter-header-wrapper') == null) {
+      const wrapper = $$('div', { className: 'coveo-filter-header-wrapper' });
+      wrapper.insertBefore(this.element);
+      return wrapper;
+    } else {
+      const wrapper = $$(document.body).find('.coveo-filter-header-wrapper');
+      return $$(wrapper);
+    }
+  }
+
+  private handleRemoveFromBreadcrumb(label: string) {
+    _.each(this.checkboxes, (object: { checkbox: Checkbox, value: string }) => {
+      if (object.value == label) {
+        object.checkbox.toggle();
+      }
+    });
+  }
+
+  // ///////////////////////////////////////BREADCRUMB//////////////////////////////////////////////
+
+  private handlePopulateBreadcrumb(args: IPopulateBreadcrumbEventArgs) {
+
+
+    let elem = document.createElement('div');
```

Comment:
`const` by default please.
Also use $$() to create element, modify class and operate on the HTML elements.

This comment applies to this whole method.